### ISSUE TITLE
Cant install on Ubuntu 16.04 LTS: node-gyp rebuild failed with ../../nan/nan.h:602:20: error: variable or field ‘AddGCEpilogueCallback’ declared `void v8::Isolate::GCEpilogueCallback` callback

### ISSUE THREAD

**Issue by npapagna** [joyent/node-bunyan-syslog#46]
Im having trouble trying to install `bunyan-syslog` in Ubuntu 16.04 LTS.
Running Node 9.4.0 and npm 5.6.0.

Reproduced the behaviour creating and empty folder, `npm init` and `npm install --save bunyan-syslog`.

Could anyone help me troubleshooting this one please?

Thanks in advance!

```
> bunyan-syslog@0.3.1 install /home/user/temp/node_modules/bunyan-syslog
> node-gyp rebuild

gyp info it worked if it ends with ok
gyp verb cli [ '/usr/local/bin/node',
gyp verb cli   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js',
gyp verb cli   'rebuild' ]
gyp info using node-gyp@3.6.2
gyp info using node@9.4.0 | linux | x64
gyp verb command rebuild []
gyp verb command clean []
gyp verb clean removing "build" directory
gyp verb command configure []
gyp verb check python checking for Python executable "python2" in the PATH
gyp verb `which` succeeded python2 /usr/bin/python2
gyp verb check python version `/usr/bin/python2 -c "import platform; print(platform.python_version());"` returned: "2.7.12\n"
gyp verb get node dir no --target version specified, falling back to host node version: 9.4.0
gyp verb command install [ '9.4.0' ]
gyp verb install input version string "9.4.0"
gyp verb install installing version: 9.4.0
gyp verb install --ensure was passed, so won't reinstall if already installed
gyp verb install version is already installed, need to check "installVersion"
gyp verb got "installVersion" 9
gyp verb needs "installVersion" 9
gyp verb install version is good
gyp verb get node dir target node version installed: 9.4.0
gyp verb build dir attempting to create "build" dir: /home/user/temp/node_modules/bunyan-syslog/build
gyp verb build dir "build" dir needed to be created? /home/user/temp/node_modules/bunyan-syslog/build
gyp verb build/config.gypi creating config file
gyp sill build/config.gypi { target_defaults: 
gyp sill build/config.gypi    { cflags: [],
gyp sill build/config.gypi      default_configuration: 'Release',
gyp sill build/config.gypi      defines: [],
gyp sill build/config.gypi      include_dirs: [],
gyp sill build/config.gypi      libraries: [] },
gyp sill build/config.gypi   variables: 
gyp sill build/config.gypi    { asan: 0,
gyp sill build/config.gypi      coverage: false,
gyp sill build/config.gypi      debug_devtools: 'node',
gyp sill build/config.gypi      debug_http2: false,
gyp sill build/config.gypi      debug_nghttp2: false,
gyp sill build/config.gypi      force_dynamic_crt: 0,
gyp sill build/config.gypi      gas_version: '2.23',
gyp sill build/config.gypi      host_arch: 'x64',
gyp sill build/config.gypi      icu_data_file: 'icudt60l.dat',
gyp sill build/config.gypi      icu_data_in: '../../deps/icu-small/source/data/in/icudt60l.dat',
gyp sill build/config.gypi      icu_endianness: 'l',
gyp sill build/config.gypi      icu_gyp_path: 'tools/icu/icu-generic.gyp',
gyp sill build/config.gypi      icu_locales: 'en,root',
gyp sill build/config.gypi      icu_path: 'deps/icu-small',
gyp sill build/config.gypi      icu_small: true,
gyp sill build/config.gypi      icu_ver_major: '60',
gyp sill build/config.gypi      llvm_version: 0,
gyp sill build/config.gypi      node_byteorder: 'little',
gyp sill build/config.gypi      node_enable_d8: false,
gyp sill build/config.gypi      node_enable_v8_vtunejit: false,
gyp sill build/config.gypi      node_install_npm: true,
gyp sill build/config.gypi      node_module_version: 59,
gyp sill build/config.gypi      node_no_browser_globals: false,
gyp sill build/config.gypi      node_prefix: '/',
gyp sill build/config.gypi      node_release_urlbase: 'https://nodejs.org/download/release/',
gyp sill build/config.gypi      node_shared: false,
gyp sill build/config.gypi      node_shared_cares: false,
gyp sill build/config.gypi      node_shared_http_parser: false,
gyp sill build/config.gypi      node_shared_libuv: false,
gyp sill build/config.gypi      node_shared_nghttp2: false,
gyp sill build/config.gypi      node_shared_openssl: false,
gyp sill build/config.gypi      node_shared_zlib: false,
gyp sill build/config.gypi      node_tag: '',
gyp sill build/config.gypi      node_use_bundled_v8: true,
gyp sill build/config.gypi      node_use_dtrace: false,
gyp sill build/config.gypi      node_use_etw: false,
gyp sill build/config.gypi      node_use_lttng: false,
gyp sill build/config.gypi      node_use_openssl: true,
gyp sill build/config.gypi      node_use_perfctr: false,
gyp sill build/config.gypi      node_use_v8_platform: true,
gyp sill build/config.gypi      node_without_node_options: false,
gyp sill build/config.gypi      openssl_fips: '',
gyp sill build/config.gypi      openssl_no_asm: 0,
gyp sill build/config.gypi      shlib_suffix: 'so.59',
gyp sill build/config.gypi      target_arch: 'x64',
gyp sill build/config.gypi      uv_parent_path: '/deps/uv/',
gyp sill build/config.gypi      uv_use_dtrace: false,
gyp sill build/config.gypi      v8_enable_gdbjit: 0,
gyp sill build/config.gypi      v8_enable_i18n_support: 1,
gyp sill build/config.gypi      v8_enable_inspector: 1,
gyp sill build/config.gypi      v8_no_strict_aliasing: 1,
gyp sill build/config.gypi      v8_optimized_debug: 0,
gyp sill build/config.gypi      v8_promise_internal_field_count: 1,
gyp sill build/config.gypi      v8_random_seed: 0,
gyp sill build/config.gypi      v8_trace_maps: 0,
gyp sill build/config.gypi      v8_use_snapshot: true,
gyp sill build/config.gypi      want_separate_host_toolset: 0,
gyp sill build/config.gypi      nodedir: '/home/user/.node-gyp/9.4.0',
gyp sill build/config.gypi      standalone_static_library: 1,
gyp sill build/config.gypi      cache_lock_stale: '60000',
gyp sill build/config.gypi      ham_it_up: '',
gyp sill build/config.gypi      legacy_bundling: '',
gyp sill build/config.gypi      sign_git_tag: '',
gyp sill build/config.gypi      user_agent: 'npm/5.6.0 node/v9.4.0 linux x64',
gyp sill build/config.gypi      always_auth: '',
gyp sill build/config.gypi      bin_links: 'true',
gyp sill build/config.gypi      key: '',
gyp sill build/config.gypi      allow_same_version: '',
gyp sill build/config.gypi      description: 'true',
gyp sill build/config.gypi      fetch_retries: '2',
gyp sill build/config.gypi      heading: 'npm',
gyp sill build/config.gypi      if_present: '',
gyp sill build/config.gypi      init_version: '1.0.0',
gyp sill build/config.gypi      user: '',
gyp sill build/config.gypi      prefer_online: '',
gyp sill build/config.gypi      force: '',
gyp sill build/config.gypi      only: '',
gyp sill build/config.gypi      read_only: '',
gyp sill build/config.gypi      cache_min: '10',
gyp sill build/config.gypi      init_license: 'ISC',
gyp sill build/config.gypi      editor: 'vi',
gyp sill build/config.gypi      rollback: 'true',
gyp sill build/config.gypi      tag_version_prefix: 'v',
gyp sill build/config.gypi      cache_max: 'Infinity',
gyp sill build/config.gypi      timing: '',
gyp sill build/config.gypi      userconfig: '/home/user/.npmrc',
gyp sill build/config.gypi      engine_strict: '',
gyp sill build/config.gypi      init_author_name: '',
gyp sill build/config.gypi      init_author_url: '',
gyp sill build/config.gypi      tmp: '/tmp',
gyp sill build/config.gypi      depth: 'Infinity',
gyp sill build/config.gypi      package_lock_only: '',
gyp sill build/config.gypi      save_dev: '',
gyp sill build/config.gypi      usage: '',
gyp sill build/config.gypi      metrics_registry: 'https://registry.npmjs.org/',
gyp sill build/config.gypi      otp: '',
gyp sill build/config.gypi      package_lock: 'true',
gyp sill build/config.gypi      progress: 'true',
gyp sill build/config.gypi      https_proxy: '',
gyp sill build/config.gypi      save_prod: '',
gyp sill build/config.gypi      cidr: '',
gyp sill build/config.gypi      onload_script: '',
gyp sill build/config.gypi      sso_type: 'oauth',
gyp sill build/config.gypi      rebuild_bundle: 'true',
gyp sill build/config.gypi      save_bundle: '',
gyp sill build/config.gypi      shell: '/bin/bash',
gyp sill build/config.gypi      dry_run: '',
gyp sill build/config.gypi      prefix: '/usr/local',
gyp sill build/config.gypi      scope: '',
gyp sill build/config.gypi      browser: '',
gyp sill build/config.gypi      cache_lock_wait: '10000',
gyp sill build/config.gypi      ignore_prepublish: '',
gyp sill build/config.gypi      registry: 'https://registry.npmjs.org/',
gyp sill build/config.gypi      save_optional: '',
gyp sill build/config.gypi      searchopts: '',
gyp sill build/config.gypi      versions: '',
gyp sill build/config.gypi      cache: '/home/user/.npm',
gyp sill build/config.gypi      send_metrics: '',
gyp sill build/config.gypi      global_style: '',
gyp sill build/config.gypi      ignore_scripts: '',
gyp sill build/config.gypi      version: '',
gyp sill build/config.gypi      local_address: '',
gyp sill build/config.gypi      viewer: 'man',
gyp sill build/config.gypi      node_gyp: '/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js',
gyp sill build/config.gypi      prefer_offline: '',
gyp sill build/config.gypi      color: 'true',
gyp sill build/config.gypi      fetch_retry_mintimeout: '10000',
gyp sill build/config.gypi      maxsockets: '50',
gyp sill build/config.gypi      offline: '',
gyp sill build/config.gypi      sso_poll_frequency: '500',
gyp sill build/config.gypi      umask: '0002',
gyp sill build/config.gypi      fetch_retry_maxtimeout: '60000',
gyp sill build/config.gypi      logs_max: '10',
gyp sill build/config.gypi      message: '%s',
gyp sill build/config.gypi      ca: '',
gyp sill build/config.gypi      cert: '',
gyp sill build/config.gypi      global: '',
gyp sill build/config.gypi      link: '',
gyp sill build/config.gypi      save: 'true',
gyp sill build/config.gypi      access: '',
gyp sill build/config.gypi      also: '',
gyp sill build/config.gypi      unicode: 'true',
gyp sill build/config.gypi      long: '',
gyp sill build/config.gypi      production: '',
gyp sill build/config.gypi      searchlimit: '20',
gyp sill build/config.gypi      unsafe_perm: 'true',
gyp sill build/config.gypi      auth_type: 'legacy',
gyp sill build/config.gypi      node_version: '9.4.0',
gyp sill build/config.gypi      tag: 'latest',
gyp sill build/config.gypi      git_tag_version: 'true',
gyp sill build/config.gypi      commit_hooks: 'true',
gyp sill build/config.gypi      script_shell: '',
gyp sill build/config.gypi      shrinkwrap: 'true',
gyp sill build/config.gypi      fetch_retry_factor: '10',
gyp sill build/config.gypi      save_exact: '',
gyp sill build/config.gypi      strict_ssl: 'true',
gyp sill build/config.gypi      dev: '',
gyp sill build/config.gypi      globalconfig: '/usr/local/etc/npmrc',
gyp sill build/config.gypi      init_module: '/home/user/.npm-init.js',
gyp sill build/config.gypi      parseable: '',
gyp sill build/config.gypi      globalignorefile: '/usr/local/etc/npmignore',
gyp sill build/config.gypi      cache_lock_retries: '10',
gyp sill build/config.gypi      searchstaleness: '900',
gyp sill build/config.gypi      node_options: '',
gyp sill build/config.gypi      save_prefix: '^',
gyp sill build/config.gypi      scripts_prepend_node_path: 'warn-only',
gyp sill build/config.gypi      group: '1000',
gyp sill build/config.gypi      init_author_email: '',
gyp sill build/config.gypi      searchexclude: '',
gyp sill build/config.gypi      git: 'git',
gyp sill build/config.gypi      optional: 'true',
gyp sill build/config.gypi      json: '' } }
gyp verb build/config.gypi writing out config file: /home/user/temp/node_modules/bunyan-syslog/build/config.gypi
gyp verb config.gypi checking for gypi file: /home/user/temp/node_modules/bunyan-syslog/config.gypi
gyp verb common.gypi checking for gypi file: /home/user/temp/node_modules/bunyan-syslog/common.gypi
gyp verb gyp gyp format was not specified; forcing "make"
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/user/temp/node_modules/bunyan-syslog/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/user/.node-gyp/9.4.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/user/.node-gyp/9.4.0',
gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/home/user/.node-gyp/9.4.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/user/temp/node_modules/bunyan-syslog',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp verb command build []
gyp verb build type Release
gyp verb architecture x64
gyp verb node dev dir /home/user/.node-gyp/9.4.0
gyp verb `which` succeeded for `make` /usr/bin/make
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/home/user/temp/node_modules/bunyan-syslog/build'
  CXX(target) Release/obj.target/syslog/src/syslog.o
In file included from ../../nan/nan.h:194:0,
                 from ../src/syslog.cc:7:
../../nan/nan_maybe_43_inl.h: In function ‘Nan::Maybe<bool> Nan::ForceSet(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Value>, v8::PropertyAttribute)’:
../../nan/nan_maybe_43_inl.h:89:64: warning: ‘v8::Maybe<bool> v8::Object::ForceSet(v8::Local<v8::Context>, v8::Local<v8::Value>, v8::Local<v8::Value>, v8::PropertyAttribute)’ is deprecated: Use CreateDataProperty / DefineOwnProperty [-Wdeprecated-declarations]
   return obj->ForceSet(GetCurrentContext(), key, value, attribs);
                                                                ^
In file included from /home/user/.node-gyp/9.4.0/include/node/v8.h:26:0,
                 from /home/user/.node-gyp/9.4.0/include/node/node.h:63,
                 from ../src/syslog.cc:5:
/home/user/.node-gyp/9.4.0/include/node/v8.h:3115:29: note: declared here
                 Maybe<bool> ForceSet(Local<Context> context, Local<Value> key,
                             ^
/home/user/.node-gyp/9.4.0/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^
In file included from ../../nan/nan.h:194:0,
                 from ../src/syslog.cc:7:
../../nan/nan_maybe_43_inl.h: In function ‘Nan::MaybeLocal<v8::Object> Nan::CloneElementAt(v8::Local<v8::Array>, uint32_t)’:
../../nan/nan_maybe_43_inl.h:221:58: warning: ‘v8::MaybeLocal<v8::Object> v8::Array::CloneElementAt(v8::Local<v8::Context>, uint32_t)’ is deprecated: Cloning is not supported. [-Wdeprecated-declarations]
   return array->CloneElementAt(GetCurrentContext(), index);
                                                          ^
In file included from /home/user/.node-gyp/9.4.0/include/node/v8.h:26:0,
                 from /home/user/.node-gyp/9.4.0/include/node/node.h:63,
                 from ../src/syslog.cc:5:
/home/user/.node-gyp/9.4.0/include/node/v8.h:3514:36: note: declared here
                 MaybeLocal<Object> CloneElementAt(Local<Context> context,
                                    ^
/home/user/.node-gyp/9.4.0/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^
In file included from ../../nan/nan_new.h:189:0,
                 from ../../nan/nan.h:200,
                 from ../src/syslog.cc:7:
../../nan/nan_implementation_12_inl.h: In static member function ‘static Nan::imp::FactoryBase<v8::BooleanObject>::return_t Nan::imp::Factory<v8::BooleanObject>::New(bool)’:
../../nan/nan_implementation_12_inl.h:40:38: warning: ‘static v8::Local<v8::Value> v8::BooleanObject::New(bool)’ is deprecated: Pass an isolate [-Wdeprecated-declarations]
   return v8::BooleanObject::New(value).As<v8::BooleanObject>();
                                      ^
In file included from /home/user/.node-gyp/9.4.0/include/node/v8.h:26:0,
                 from /home/user/.node-gyp/9.4.0/include/node/node.h:63,
                 from ../src/syslog.cc:5:
/home/user/.node-gyp/9.4.0/include/node/v8.h:4875:56: note: declared here
   V8_DEPRECATED("Pass an isolate", static Local<Value> New(bool value));
                                                        ^
/home/user/.node-gyp/9.4.0/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^
In file included from ../src/syslog.cc:7:0:
../../nan/nan.h: At global scope:
../../nan/nan.h:602:20: error: variable or field ‘AddGCEpilogueCallback’ declared void
       v8::Isolate::GCEpilogueCallback callback
                    ^
../../nan/nan.h:602:7: error: ‘GCEpilogueCallback’ is not a member of ‘v8::Isolate’
       v8::Isolate::GCEpilogueCallback callback
       ^
../../nan/nan.h:603:18: error: expected primary-expression before ‘gc_type_filter’
     , v8::GCType gc_type_filter = v8::kGCTypeAll) {
                  ^
../../nan/nan.h:608:20: error: variable or field ‘RemoveGCEpilogueCallback’ declared void
       v8::Isolate::GCEpilogueCallback callback) {
                    ^
../../nan/nan.h:608:7: error: ‘GCEpilogueCallback’ is not a member of ‘v8::Isolate’
       v8::Isolate::GCEpilogueCallback callback) {
       ^
../../nan/nan.h:613:20: error: variable or field ‘AddGCPrologueCallback’ declared void
       v8::Isolate::GCPrologueCallback callback
                    ^
../../nan/nan.h:613:7: error: ‘GCPrologueCallback’ is not a member of ‘v8::Isolate’
       v8::Isolate::GCPrologueCallback callback
       ^
../../nan/nan.h:614:18: error: expected primary-expression before ‘gc_type_filter’
     , v8::GCType gc_type_filter = v8::kGCTypeAll) {
                  ^
../../nan/nan.h:619:20: error: variable or field ‘RemoveGCPrologueCallback’ declared void
       v8::Isolate::GCPrologueCallback callback) {
                    ^
../../nan/nan.h:619:7: error: ‘GCPrologueCallback’ is not a member of ‘v8::Isolate’
       v8::Isolate::GCPrologueCallback callback) {
       ^
../../nan/nan.h: In function ‘bool Nan::SetAccessor(v8::Local<v8::Object>, v8::Local<v8::String>, Nan::GetterCallback, Nan::SetterCallback, v8::Local<v8::Value>, v8::AccessControl, v8::PropertyAttribute)’:
../../nan/nan.h:1961:16: warning: ‘bool v8::Object::SetAccessor(v8::Local<v8::Name>, v8::AccessorNameGetterCallback, v8::AccessorNameSetterCallback, v8::Local<v8::Value>, v8::AccessControl, v8::PropertyAttribute)’ is deprecated: Use maybe version [-Wdeprecated-declarations]
     , attribute);
                ^
In file included from /home/user/.node-gyp/9.4.0/include/node/v8.h:26:0,
                 from /home/user/.node-gyp/9.4.0/include/node/node.h:63,
                 from ../src/syslog.cc:5:
/home/user/.node-gyp/9.4.0/include/node/v8.h:3184:22: note: declared here
                 bool SetAccessor(Local<Name> name,
                      ^
/home/user/.node-gyp/9.4.0/include/node/v8config.h:318:3: note: in definition of macro ‘V8_DEPRECATED’
   declarator __attribute__((deprecated(message)))
   ^
syslog.target.mk:95: recipe for target 'Release/obj.target/syslog/src/syslog.o' failed
make: *** [Release/obj.target/syslog/src/syslog.o] Error 1
make: Leaving directory '/home/user/temp/node_modules/bunyan-syslog/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:160:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:12)
gyp ERR! System Linux 4.13.0-32-generic
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/user/temp/node_modules/bunyan-syslog
gyp ERR! node -v v9.4.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
npm WARN temp@1.0.0 No description
npm WARN temp@1.0.0 No repository field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! bunyan-syslog@0.3.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the bunyan-syslog@0.3.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

```

**Comment by melloc** [comment]
@npapagna Hi, I've just published `bunyan-syslog@0.3.2`, which uses a newer version of `nan`. This should resolve your issues. Please reopen this ticket if this is not the case (or a new ticket if you get a different error.)

**Comment by gijo-varghese** [comment]
This issue still exists!
